apps/emqx_authn/src/emqx_authn_api.erl

```diff
@@ -1267,6 +1303,9 @@ paginated_list_type(Type) ->
         {meta, ref(emqx_dashboard_swagger, meta)}
     ].
 
+tenant(_Req = #{headers := Headers}) ->
+    maps:get(<<"emqx_tenant">>, Headers, undefined).
```

Comment:
please make sure the headers are normalised by cowboy
```suggestion
tenant(_Req = #{headers := Headers}) ->
    maps:get(<<"emqx-tenant-id">>, Headers, undefined).
```

Reply:
I still think a safer alternative is the add the tenant ID to the token to simplify the auth.
Even if for now it seems that all API calls will be issued by a deployment agent which makes sure the header is added for the intended tenant, but it feels like a security whole.